src/Stryker.Core/Stryker.Core/Initialisation/InitialTestProcess.cs

```diff
@@ -3,55 +3,79 @@
+    }
+
     public interface IInitialTestProcess
     {
-        int InitialTest(ITestRunner testRunner);
-        int TotalNumberOfTests { get; }
+        InitialTestRun InitialTest(IStrykerOptions options, ITestRunner testRunner);
     }
 
     public class InitialTestProcess : IInitialTestProcess
     {
         private readonly ILogger _logger;
-        public int TotalNumberOfTests { get; private set; }
+        private TestRunResult _initTestRunResult;
 
         public InitialTestProcess()
         {
             _logger = ApplicationLogging.LoggerFactory.CreateLogger<InitialTestProcess>();
         }
 
+        public TestRunResult InitialTestRun => _initTestRunResult;
+        public ITimeoutValueCalculator TimeoutValueCalculator { get; private set; }
+
         /// <summary>
         /// Executes the initial testrun using the given testrunner
         /// </summary>
         /// <param name="testRunner"></param>
+        /// <param name="options">Stryker options</param>
         /// <returns>The duration of the initial testrun</returns>
-        public int InitialTest(ITestRunner testRunner)
+        public InitialTestRun InitialTest(IStrykerOptions options, ITestRunner testRunner)
         {
-            var message = testRunner.DiscoverNumberOfTests() is var total && total == -1 ? "Unable to detect" : $"{total}";
+            var message = testRunner.DiscoverTests() is var total && total.Count == 0 ? "Unable to detect" : $"{total.Count}";
 
-            TotalNumberOfTests = total;
-            _logger.LogInformation("Total number of tests found: {0}", message);
+            _logger.LogInformation("Total number of tests found: {0}.", message);
 
-            _logger.LogInformation("Initial testrun started");
+            _logger.LogInformation("Initial testrun started.");
 
             // Setup a stopwatch to record the initial test duration
             var stopwatch = new Stopwatch();
             stopwatch.Start();
 
-            var testResult = testRunner.RunAll(null, null, null);
+            _initTestRunResult = testRunner.InitialTest();
             // Stop stopwatch immediately after testrun
             stopwatch.Stop();
 
-            _logger.LogDebug("Initial testrun output: {0}", testResult.ResultMessage);
-            if (testResult.FailingTests.Count > 0)
+            // timings
+            _logger.LogDebug("Initial testrun output: {0}.", _initTestRunResult.ResultMessage);
+            if (!_initTestRunResult.FailingTests.IsEmpty)
             {
-                _logger.LogWarning("Initial test run failed. Mutation score cannot be computed.");
-                throw new StrykerInputException("Initial testrun was not successful.", testResult.ResultMessage);
+                var failingTestsCount = _initTestRunResult.FailingTests.Count;
+                _logger.LogWarning($"{failingTestsCount} tests are failing.");
+                if (((double)failingTestsCount) / _initTestRunResult.RanTests.Count > .1)
```

Comment:
Technically, we need just one test, but if several tests are failing, this is something that could be fixed before dealing with mutants. 
More often than not, if you many failing tests, this is a sure sign that something went wrong during the build or test phase.
So I feel this should be a low tolerance by default, but it could configurable within the new option system, I guess.

but ultimately, this could be any number you are confortable with

Reply:
I'd like to see at least 50%. We don't know the context, why some tests are failing. We print the warning so they can think for themselves if they want to continue or break up the run.